math/Frame3f.cs

```diff
@@ -241,170 +241,170 @@ public Vector2f ToPlaneUV(Vector3f p, int nNormal, int nAxis0 = -1, int nAxis1 =
 
 
         ///<summary> distance from p to frame-axes-plane perpendicular to normal axis </summary>
-        public float DistanceToPlane(Vector3f p, int nNormal)
+        public readonly float DistanceToPlane(Vector3f p, int nNormal)
         {
             return Math.Abs((p - origin).Dot(GetAxis(nNormal)));
         }
         ///<summary> signed distance from p to frame-axes-plane perpendicular to normal axis </summary>
-		public float DistanceToPlaneSigned(Vector3f p, int nNormal)
+		public readonly float DistanceToPlaneSigned(Vector3f p, int nNormal)
 		{
 			return (p - origin).Dot(GetAxis(nNormal));
 		}
 
 
         ///<summary> Map point *into* local coordinates of Frame </summary>
-		public Vector3f ToFrameP(Vector3f v) {
+		public readonly Vector3f ToFrameP(Vector3f v) {
             v.x -= origin.x; v.y -= origin.y; v.z -= origin.z;
             return rotation.InverseMultiply(ref v);
         }
         ///<summary> Map point *into* local coordinates of Frame </summary>
-		public Vector3f ToFrameP(ref Vector3f v) {
+		public readonly Vector3f ToFrameP(ref Vector3f v) {
```

Comment:
I suggest that we use `in` instead of `ref` where possible.

Reply:
"ref" existed in this API for long, so these methods already used somewhere with "ref" keyword, so we cannot remove "ref" now.